pkg/minishift/network/settings.go

```diff
@@ -312,7 +312,7 @@ func WriteNetworkSettingsToInstance(driver drivers.Driver, networkSettings Netwo
 
 // CheckInternetConnectivity return false if user is not connected to internet
 func CheckInternetConnectivity(address string) bool {
-	_, err := net.Dial("tcp", address)
+	_, err := http.Get(address)
```

Comment:
This is really the most meat of this commit as `http.Get` actually knows about proxies set in the environment.